tests/unit/sharded_copying_manager_tests/common.py

```diff
@@ -0,0 +1,919 @@
+        for req in request:
+            lines = extract_lines_from_request(req)
+            result.append(lines)
+
+        return list(itertools.chain(*result))
+
+    # extract single request.
+    parsed_request = test_util.parse_scalyr_request(request.get_payload())
+
+    lines = []
+
+    if "events" in parsed_request:
+        for event in parsed_request["events"]:
+            if "attrs" in event:
+                attrs = event["attrs"]
+                if "message" in attrs:
+                    lines.append(attrs["message"].strip())
+
+    return lines
+
+
+class CopyingManagerCommonTest(object):
+    """
+    Test case with many helpful features for CopyingManager and CopyingManagerWorker testing.
+    """
+
+    def setup(self):
+        self._config_builder = None  # type: Optional[ConfigBuilder]
+        self._instance = None
+
+    def teardown(self):
+        if self._config_builder is not None:
+            self._config_builder.clear()
+
+    def _extract_lines(self, request):
+        return extract_lines_from_request(request)
+
+    def _create_config(
+        self, log_files_number=1, use_pipelining=False, config_data=None
+    ):
+        pipeline_threshold = 1.1
+        if use_pipelining:
+            pipeline_threshold = 0.0
+
+        config_initial_data = {
+            "debug_level": 5,
+            "disable_max_send_rate_enforcement_overrides": True,
+            "pipeline_threshold": pipeline_threshold,
+        }
+
+        if config_data is not None:
+            config_initial_data.update(config_data)
+
+        test_files, self._config_builder = ConfigBuilder.build_config_with_n_files(
+            log_files_number, config_data=config_initial_data
+        )
+
+    def _append_lines(self, *lines, **kwargs):
+        # type: (*str, **TestableLogFile) -> None
+        log_file = kwargs.get(six.ensure_str("log_file"))
+        if log_file is None:
+            if self._current_log_file is not None:
+                log_file = self._current_log_file
+            else:
+                raise RuntimeError("File is not specified.")
+
+        log_file.append_lines(*lines)
+
+    def _wait_for_rpc(self):
+        """
+        Wraps the next 'self._manager.controller.wait_for_rpc()
+        and returns lines instead' of requests.
+        """
+        (request, responder_callback) = self._instance.wait_for_rpc()
+        request_lines = self._extract_lines(request)
+        return request_lines, responder_callback
+
+    def _wait_for_rpc_and_respond(self, response="success"):
+        """
+        Wraps the next 'self._manager.controller.wait_for_rpc(), returns lines instead' of requests alongside with
+        response  callback.
+        :param response: response for request.
+        """
+        (request_lines, responder_callback) = self._wait_for_rpc()
+        responder_callback(response)
+
+        return request_lines
+
+    def _append_lines_with_callback(self, *lines, **kwargs):
```

Comment:
Is there any particular reason you aren't explicitly listing the kwargs here?  I'd rather we list them if possible because it improves readability.

Also, can you provide documentation here.  It is not immediately clear to me what the callback is used for.